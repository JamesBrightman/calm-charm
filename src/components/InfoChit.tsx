import React, { FC, HTMLAttributes } from 'react';

export type InfoChitProps = HTMLAttributes<HTMLDivElement>

export const InfoChit: FC<InfoChitProps> = ({
  children,
  className,
  ...props
}) => {
  return (
    <div
      className={
        'flex flex-col gap-1 bg-stone-200 p-2 rounded-lg text-gray-700' +
        ' ' +
        className
      }
      {...props}
    >
      {children}
    </div>
  );
};
