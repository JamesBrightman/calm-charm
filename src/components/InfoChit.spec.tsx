import React from 'react';
import { render, screen } from '@testing-library/react';
import { InfoChit } from './InfoChit';

describe('Info Chit Component', () => {
  it('Info chit renders', () => {
    const { baseElement } = render(
      <InfoChit>
        <p>{'Foo Bar'}</p>
      </InfoChit>
    );

    expect(baseElement).toBeTruthy();
  });

  it('Info chit matches snapshot', () => {
    const { baseElement } = render(
      <InfoChit>
        <p>{'Foo Bar'}</p>
      </InfoChit>
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('Info Chit displays custom text', () => {
    render(
      <InfoChit>
        <p>{'Bar Foo'}</p>
      </InfoChit>
    );

    expect(screen.getByText('Bar Foo')).toBeInTheDocument();
  });

  it('Info Chit renders with custom class', () => {
    render(
      <InfoChit className="bg-blue-500 text-lg" data-testId="chitTest">
        <p>{'Foo Bar'}</p>
      </InfoChit>
    );

    expect(screen.getByTestId('chitTest')).toHaveClass('bg-blue-500 text-lg');
  });
});
