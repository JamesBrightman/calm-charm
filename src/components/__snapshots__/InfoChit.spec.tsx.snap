// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info Chit Component Info chit matches snapshot 1`] = `
<body>
  <div>
    <div
      class="flex flex-col gap-1 bg-stone-200 p-2 rounded-lg text-gray-700 undefined"
    >
      <p>
        Foo Bar
      </p>
    </div>
  </div>
</body>
`;
